transport/split/stream_dialer.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package split
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type splitDialer struct {
+	dialer     transport.StreamDialer
+	splitPoint int64
+}
+
+var _ transport.StreamDialer = (*splitDialer)(nil)
+
+// NewStreamDialer creates a client that splits the outgoing strean at byte splitpoint.
```

Comment:
I think we need to clarify the exact byte index of `splitPoint`. Like if `splitPoint = 1`, does it mean `[0] |<-splitPoint [1] [2] ... ` or `[0] [1] |<-splitPoint [2] ... ` or something else?

```suggestion
// NewStreamDialer creates a [transport.StreamDialer] that splits the outgoing stream at byte splitPoint (starting from byte 1).
```

Reply:
I renamed `splitPoint` to `prefixBytes` and updated the comment. I hope that helps.